### PR TITLE
Automate enhancement on the perf data collection for JIT CodeQuality benchmarks on Windows.

### PR DESCRIPTION
**Pull request by smile21prc**
1. Enable machine data collection.
2. Upload generated machinedata.json and all perf-*.xml files to /artifact/ Jenkins dir.
3. Enable InstRetired.
4. Force the perf runs to use physical machines.
5. Add test name, and benchmark InstRetired to the console output per Eugene's request.
6. Scripts under tests/scripts/benchview and tests/scripts/dependencies are minimal set of standard http://benchview dependency scripts to make machinedata.py work well on Jenkins machine.

Latest Jenkins run using physical machine passes with log at:
http://dotnet-ci2.cloudapp.net/job/smile21prc_coreclr/job/coreclr-perf/job/perf_Windows_NT/24/consoleFull

### CODE REVIEW

**Comment by smile21prc** [comment]
@jashook @erozenfeld

File: perf.groovy [hunk 1]
```diff
@@ -6,20 +6,28 @@ def project = GithubProject
...
                     // Batch
+                    batchFile("python tests\\scripts\\machinedata.py")
                     batchFile("set __TestIntermediateDir=int&&build.cmd release x64")
                     batchFile("tests\\runtest.cmd release x64")
                     batchFile("tests\\scripts\\run-xunit-perf.cmd")
             }
         }
 
-        Utilities.setMachineAffinity(newJob, os, 'latest-or-auto-elevated') // Just run against Windows_NT VM�s for now.
+        //Utilities.setMachineAffinity(newJob, os, 'latest-or-auto-elevated') // Disable to forcely use physical machine.
```
Comment (jashook) [inline perf.groovy]: There is no reason to keep this commented in the file

**Comment by jashook** [comment]
Instead of checking in these python tools can we pull them from the Nuget package?

File: perf.groovy (hunk 1 above)
Comment (smile21prc) [inline perf.groovy]: Sure, will remove it.

**Comment by smile21prc** [comment]
@jashook Good suggestion. Will try it out on the Jenkins machine, and pull them from the Nuget package instead if they are accessible.

**Comment by smile21prc** [comment]
@jashook , I've removed the suggest commented line, and changed the script to pull uploading python files from the Nuget package instead of checking them in, which can be seen from the logs of latest successful run:
http://dotnet-ci2.cloudapp.net/job/smile21prc_coreclr/job/coreclr-perf/job/perf_Windows_NT/27/consoleFull

And none of those three checking failures are related to my changes.

Does it now look good to you for the merge?

**Comment by jashook** [comment]
lgtm

**Comment by smile21prc** [comment]
@jashook  
Thanks Jarret.

**Pull request merged**